app/src/main/java/com/example/android/oscarine/signin/SigninFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.android.oscarine.signin
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.example.android.oscarine.databinding.FragmentSigninBinding
+
+class SigninFragment : Fragment() {
+
+    private val viewModel: SigninViewModel by lazy {
+        ViewModelProviders.of(this).get(SigninViewModel::class.java)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val binding = FragmentSigninBinding.inflate(inflater)
+
+        @Suppress("UsePropertyAccessSyntax")
+        binding.setLifecycleOwner(this)
+        binding.viewModel = viewModel
+
+        binding.loginSubmitButton.setOnClickListener {
+            viewModel.loginSubmitButtonClicked()
+        }
+        viewModel.response.observe(this, Observer {
+            if (it == "200") {
```

Comment:
Don't add logic in views

Reply:
On the basis of this logic, I am showing the Toast to the user for which I have to observe `response` from the view model. I think this much logic is allowed and I have learned about this from Udacity's course. What else can I do if not this?

Reply:
This logic should be present in the ViewModel. Only view rendering logic should be present in fragments